adoptingerlang.org

```diff
@@ -4101,20 +4101,118 @@ With these broad lines covered, we can start exploring the processes and practic
+
+Uncool languages or languages that are "hard to hire for" can always be a problem. Unfortunately, this is a problem you can _never_ actually solve, because nobody knows the future. Is it worse to have a single main language people stop liking and then none of your stack is easy to staff, or to have only half of it being hard to hire for? What about cases like Python, where the migration from 2.x to 3.x has been so hard that libraries and components essentially consider them to be distinct languages? We consider that organisations that do not marry themselves to a single language but instead make a habit of switching between a few of them (and not just a ton of them) train themselves to better adjust to variations in the market of programming languages.
+
+Then we have the complaint about the lone developer not caring about what the organisation does or knows, and who starts shipping things in a language nobody else knows. That developer might be doing this as part of resume padding, and then leave everyone behind after hopping to a new job. There's no protection against that, even when you mandate a single language. Hell, a developer could do the same by using your company's main language but by trying a new framework or new paradigm supported by the language or one of its libraries. Having a single unsupported [[https://aws.amazon.com/lambda/][Lambda]] could be trickier than a new language within a known deployment context.
+
+We mostly consider this negative behaviour, letting such work take place in complete isolation, a failure of process that must be addressed, rather than blaming the tool for being used.
+
+Finally, there's one major argument to undo. It's a commonly accepted idea that having many languages creates barriers, and that you will have an easier time moving developers around if you use one language and one framework. In fact, we'll call this the biggest myth against polyglot organisations.
+
+It's misguided because it assumes developers are like mechanical parts. As long as the interface is the same, they can be interchangeable. It ignores everything not related to the language. Different teams have different habits and dynamics you need to get used to. They work on projects with different histories and challenges, in possibly distinct domains, with different focuses and compromises done in how they engineer things. To add to this, even within a single language, there is a cost of migration: different frameworks, libraries, toolchains, or versions of these toolchains all require some adaptation.
```

Comment:
I like this point!  Domain knowledge and engineering standards mostly trump tooling.